plugin.video.wizjatv/main.py

```diff
@@ -33,55 +36,38 @@ def WizjaTV():
     s, content = obj.ListaKanalow()
     content = json.loads(content)
     for item in content:
-        addDir(item['title'],item['url'] , 6, item['icon'],'','', False)
+        addDir(name=item['title'], url=item['url'], mode=PLAY_LINK, thumb=item['icon'], isFolder=False)
 
-def OdpalanieLinku():
+def OdpalanieLinku(url):
     try:
         global s
-        url = urllib.unquote_plus(params['url'])
         s, content = obj.ListaKanalow()
-        link = obj.Link(url,s)
+        link = obj.Link(url, s)
         xbmc.Player().play(str(link).replace("rtmp://$OPT:rtmp-raw=", ""))
     except:
         log_exception()
-        pass
 
 def get_params():
     paramstring = sys.argv[2]
     if paramstring.startswith('?'):
         paramstring = paramstring[1:]
-    return dict((k, vv[0]) for k, vv in parse_qs(paramstring).items())
+    return dict((k, urllib.unquote_plus(vv[0])) for k, vv in parse_qs(paramstring).items())
 
 params = get_params()
-url = None
-name = None
-thumb = None
-mode = None
-iconimage = None
+url = params.get('url')
+name = params.get('name')
+iconimage = params.get('iconimage')
 
-try:
-    url = urllib.unquote_plus(params['url'])
-except:
-    pass
-try:
-    name = params.get('name')
-except:
-    pass
 try:
     mode = int(params.get('mode'))
-except:
+except (TypeError, ValueError):
```

Comment:
To też jest istotne. Należy zwężać zasięg except tylko do przewidzianych wyjątków. Dzięki temu coś niespodziewanego się wysypie i będziemy mieli okazję to znaleźć.
Przy okazji wcześniej pisałem o ValueError (np. dla `int('ala')`) a zapomniałem o TypeError (np. dla `int(None)`).